src/cpu/o3/rename.cc

```diff
@@ -1225,9 +1225,6 @@ Rename::checkStall(ThreadID tid)
     } else if (calcFreeLQEntries(tid) <= 0 && calcFreeSQEntries(tid) <= 0) {
         DPRINTF(Rename,"[tid:%i] Stall: LSQ has 0 free entries.\n", tid);
         ret_val = true;
-    } else if (renameMap[tid]->numFreeEntries() <= 0) {
```

Comment:
Could you explain why this stall case is no longer needed? I assume that the original intention of this check (something like `renameMaps[<inst type>].numFreeEntries()`) is checked somewhere else, but I'm not immediately seeing it.

Reply:
Hey Jason, I describe it in the commit message: https://github.com/gem5/gem5/pull/828/commits/f53a2bf9d3be38dd60c90373de6891265ad97f95. If it is still not clear I can expand, please let me know and thanks for the review

Reply:
In short numFreeEntries is a bit misleading as it does not differentiate according to <inst type>. It just returns the *minimum* number of freeList registers across all class. For example, if I have

IntRegs -> 200 free regs
FloatRegs -> 100 free regs
EsoticRegClass -> 0 free regs.

Then rename will allways stall even if there is no instruction in the pipeline making use of EsoticRegClass registers! This is overly conservative, rename should be stopped only if instructions within the pipeline can't be renamed (and this happen in renameInsts: https://github.com/gem5/gem5/blob/stable/src/cpu/o3/rename.cc#L662

Reply:
Ah! Got it. This makes sense.